mat/svd.go

```diff
@@ -20,6 +20,33 @@ type SVD struct {
 	vt blas64.General
 }
 
+// SVDKind specifies the treatment of singular vectors during an SVD
+// factorization.
+type SVDKind int
+
+const (
+	// SVDNone specifies that no singular vectors should be computed during
+	// the decomposition.
+	SVDNone SVDKind = 0
+
+	SVDThinU SVDKind = 1 << (iota - 1)
+	SVDFullU
+	SVDThinV
+	SVDFullV
+
+	// SVDThin computes the thin singular vectors, that is, it computes
```

Comment:
I understand that this is a shortcut but it suggests as if there are two kinds singular vectors, thin and non-thin. I think there is only a single kind of (ok, left and right) singular vectors. In the thin case only the singular vectors are computed while in the full case the set of singular vectors (in the first min(m,n) columns of U and V) is extended to form a complete orthonormal basis of R^n.

Reply:
Is that a statement about naming, behavior, or documentation? There is only one (non-unique) set of eigenvalues, but the factorized matrix has rank at most min(m,n). The Full decomposition returns all of the vectors. The thin decomposition only computes the ones in the rank of A. This is pretty commonly called the thin decomposition.

Reply:
About documentation. It would be enough to just replace `thin singular vectors` with `thin singular value decomposition and maybe below add `The columns of U~ and V~ contain the left and right singular vectors, respectively.`